python/graphstorm/model/lp_gnn.py

```diff
@@ -154,6 +154,37 @@ def lp_mini_batch_predict(model, emb, loader, device):
             Rankings of positive scores in format of {etype: ranking}
     """
     decoder = model.decoder
+    return run_lp_mini_batch_predict(decoder,
+                                     emb,
+                                     loader,
+                                     device)
+
+def run_lp_mini_batch_predict(decoder, emb, loader, device):
+    """ Perform mini-batch link prediction with the given decoder.
+
+        This function follows full-graph GNN embedding inference.
+        After having the GNN embeddings, we need to perform mini-batch
+        computation to make predictions on the GNN embeddings.
+
+        Note: caller should call model.eval() before calling this function
```

Comment:
Should we add model.eval() between line #156 and #157, and add model.train() after calling this method in the line #157?

Reply:
In lp_mini_batch_predict, we only use the model decoder, so we do not know whether the caller will do something else for evaluation after calling lp_mini_batch_predict.

Let me add some notes in the function.

Reply:
I C.